projects/components/src/table/controls/table-controls-api.ts

```diff
@@ -1,58 +1,77 @@
 import { Dictionary } from '@hypertrace/common';
-import { SelectOption } from '../../select/select-option';
+import { FilterOperator } from '../../filtering/filter/filter-operators';
 import { TableFilter } from '../table-api';
 
-export interface SelectControl {
-  placeholder?: string;
-  default?: SelectOption<TableControlOption>;
-  options: SelectOption<TableControlOption>[];
+export const enum TableControlOptionType {
+  Filter = 'filter',
+  Property = 'property',
+  Unset = 'unset'
 }
 
-export interface SelectChange {
-  select: SelectControl;
-  value: TableControlOption;
-}
+export type TableControlOption = TableUnsetControlOption | TableFilterControlOption | TablePropertyControlOption;
 
-export interface CheckboxControl {
+export interface TableFilterControlOption {
+  type: TableControlOptionType.Filter;
   label: string;
-  value: boolean;
-  options: TableCheckboxOptions;
+  metaValue: TableFilter;
 }
 
-export interface CheckboxChange {
-  checkbox: CheckboxControl;
-  option: TableControlOption<boolean>;
+export interface TableUnsetControlOption {
+  type: TableControlOptionType.Unset;
+  label: string;
+  metaValue: string;
 }
 
-export const enum TableControlOptionType {
-  Filter = 'filter',
-  Property = 'property',
-  UnsetFilter = 'unset-filter'
+export interface TablePropertyControlOption {
+  type: TableControlOptionType.Property;
+  label: string;
+  metaValue: Dictionary<unknown>;
 }
 
-export type TableControlOption<T = unknown> =
-  | TableUnsetFilterControlOption<T>
-  | TableFilterControlOption<T>
-  | TablePropertyControlOption<T>;
+/*
+ * Select Control
+ */
 
-interface TableControlOptionBase<T> {
-  value?: T;
+export interface SelectControl {
+  placeholder: string;
+  options: TableSelectControlOption[];
 }
-export interface TableUnsetFilterControlOption<T = unknown> extends TableControlOptionBase<T> {
-  type: TableControlOptionType.UnsetFilter;
-  metaValue: string;
+
+export interface SelectChange {
+  select: SelectControl;
+  values: TableSelectControlOption[];
 }
 
-export interface TableFilterControlOption<T = unknown> extends TableControlOptionBase<T> {
-  type: TableControlOptionType.Filter;
-  metaValue: TableFilter;
+export type TableSelectControlOption = TableFilterControlOption;
+
+/*
+ * Checkbox Control
+ */
+
+export interface CheckboxControl {
+  label: string;
+  value: boolean;
+  options: TableCheckboxOptions;
 }
 
-export interface TablePropertyControlOption<T = unknown> extends TableControlOptionBase<T> {
-  type: TableControlOptionType.Property;
-  metaValue: Dictionary<unknown>;
+export interface CheckboxChange {
+  checkbox: CheckboxControl;
+  option: TableCheckboxControlOption;
 }
 
-export type TableCheckboxControlOption<T extends boolean> = TableControlOption<T> & { label: string };
+export type TableCheckboxControlOption<T extends boolean = boolean> = TableControlOption & {
```

Comment:
T extends boolean, a primitive? why can't it be just a boolean?

Reply:
ok so this is the usage. We can just keep it as boolean right?
```
export type TableCheckboxOptions = [TableCheckboxControlOption<true>, TableCheckboxControlOption<false>]
```

Reply:
Yeah, also strange, but the `extends` is required for us to do line 66.

Reply:
if we keep boolean why wouldn't line 66 work?

Reply:
Hmm. Trying again, removing `extends` actually works now that I am setting a default. I wasn't before and only recently added it. Update incoming.

Reply:
cool